File: eco-core/core-proxy/src/main/java/com/willfp/eco/proxy/proxies/SkullProxy.java
package com.willfp.eco.proxy.proxies;

import com.willfp.eco.core.proxy.AbstractProxy;
import org.bukkit.inventory.meta.SkullMeta;
import org.jetbrains.annotations.NotNull;

public interface SkullProxy extends AbstractProxy {
    /**
     * Set the texture of a skull from base64.
     *
     * @param meta   The meta to modify.
     * @param base64 The base64 texture.
     */
    void setSkullTexture(@NotNull SkullMeta meta,
                         @NotNull String base64);
}

File: eco-core/core-proxy/src/main/kotlin/com/willfp/eco/internal/spigot/proxy/SkullProxy.kt
package com.willfp.eco.internal.spigot.proxy

import org.bukkit.inventory.meta.SkullMeta

interface SkullProxy {
    fun setSkullTexture(
        meta: SkullMeta,
        base64: String
    )

    fun getSkullTexture(
        meta: SkullMeta
    ): String?
}